proto/substrait/algebra.proto

```diff
@@ -884,6 +885,12 @@ message Expression {
       int32 microseconds = 3;
     }
 
+    message IntervalMonthToNano {
+      int32 months = 1;
+      int32 days = 2;
+      int64 nanoseconds = 3;
```

Comment:
I opted for nanoseconds here for simplicity, but alternative is to use a variable precision instead like PrecisionTimestamp(Tz) - preferences?